src/azul/drs.py

```diff
@@ -131,7 +132,8 @@ def _get_object(self, drs_uri: str, access_method: AccessMethod) -> Access:
                 else:
                     require(furl(access_url['url']).scheme == access_method.scheme)
                     return Access(method=access_method,
-                                  url=access_url['url'])
+                                  url=access_url['url'],
+                                  access_id=method['access_id'])
```

Comment:
I want this to be as generic a TDR client as possible. The fact that TDR yields both `url` and `acccess_id` is (in my opinion) a quirk of their particular implementation. I would expect a sane implementations to return either one or the other but not both. We should NOT rely on access_id being present alongside the URL but instead test for it. We also don't want to expose access_id to the caller or put the burden of resolving it on them. As in the `if` branch above, the access id should be resolved here. We do need to work around the TDR quirks but when and where we do, we need to make that obvious with comments.

The `access_id` parameter to `get_object` should be eliminated. I don't think it is currently used on `develop`.

Reply:
These changes are not in a fixup commit because separating them from the original commit caused conflicts with the new commit I added for removing the `access_id` parameter.